### PR TITLE
Tracing and opentelemetry integration

### PR DESCRIPTION
**Pull request by alex-hunt-materialize**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -165,19 +185,20 @@ async fn set_eip_status_created(
     let params = PatchParams::default();
     let result = eip_api.patch_status(eip_name, &params, &patch).await;
     if result.is_ok() {
-        info!("Done updating status");
+        event!(Level::INFO, "Done updating status for created EIP.");
```
Comment (antifuchs) [inline src/main.rs:188]: Thankfully in a tracing world, I think these kinds of log statements are kinda redundant.

**Review by antifuchs — APPROVED** [review]
Looks great to me! I just hope we don't rely on vendored/otherwise-pinned GRPC system libs

**Review by zpu-materialize — APPROVED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 FROM rust:1.56.1-slim-buster as builder
-RUN apt-get update && apt-get -qy install pkg-config libssl-dev && apt-get clean
+RUN apt-get update && apt-get -qy install pkg-config libssl-dev cmake g++ && apt-get clean
```
[the remark this answers is not in the document]
Comment (alex-hunt-materialize) [reply Dockerfile:2]: I don't see any way to use a non-vendored GRPC with grpc-sys. It doesn't seem to have any feature flag for it.

File: src/main.rs (hunk 1 above)
Comment (alex-hunt-materialize) [reply src/main.rs:188]: Mostly. They are still nice for the stdout.

**Pull request merged**